File: ShoppingList.Angular/src/app/shared/recipe-part.model.ts
import { Ingredient } from './ingredient.model';

export class RecipePart {
    id: string;
    ingredient: Ingredient;
    unit: string;
    quantity: number;

    constructor(ingredient: Ingredient, unit: string, quantity: number) {
        this.ingredient = ingredient;
        this.unit = unit;
        this.quantity = quantity;
    }
}

File: RecipeBook.Business/Models/RecipePartModel.cs
using System;

namespace RecipeBook.Business.Models
{
    public class RecipePartModel
    {
        public Guid? Id { get; set; }
        public IngredientModel Ingredient { get; set; }
        public string Unit { get; set; }
        public decimal Quantity { get; set; }
    }
}
